rs/execution_environment/src/query_handler/query_cache/tests.rs

```diff
@@ -1563,3 +1567,148 @@ fn query_cache_future_proof_test() {
+    const MORE_HEAP_BYTES: usize = 3;
+
+    // Key with no heap data.
+    let key = EntryKey {
+        source: user_test_id(1),
+        receiver: CanisterId::from_u64(1),
+        method_name: String::new(),
+        method_payload: vec![],
+    };
+    assert_eq!(size_of_val(&key), 112);
+    assert_eq!(key.memory_bytes(), size_of_val(&key));
+
+    // Key with some heap data.
+    let key = EntryKey {
+        source: user_test_id(1),
+        receiver: CanisterId::from_u64(1),
+        method_name: " ".repeat(HEAP_BYTES),
+        method_payload: vec![42; HEAP_BYTES],
+    };
+    assert_eq!(size_of_val(&key), 112);
+    assert_eq!(key.memory_bytes(), size_of_val(&key) + HEAP_BYTES * 2);
+
+    // Key with more heap data.
+    let key = EntryKey {
+        source: user_test_id(1),
+        receiver: CanisterId::from_u64(1),
+        method_name: "0".repeat(MORE_HEAP_BYTES),
+        method_payload: vec![42; MORE_HEAP_BYTES],
+    };
+    assert_eq!(size_of_val(&key), 112);
+    assert_eq!(key.memory_bytes(), size_of_val(&key) + MORE_HEAP_BYTES * 2);
+
+    // Value with no heap data.
+    let env = EntryEnv {
+        batch_time: time::GENESIS,
+        canisters_versions_balances_stats: vec![],
+    };
+    let value = EntryValue::new(
+        env,
+        Result::Ok(WasmResult::Reply(vec![])),
+        &SystemApiCallCounters::default(),
+    );
+    assert_eq!(size_of_val(&value), 80);
+    assert_eq!(value.memory_bytes(), size_of_val(&value));
+
+    // Value with some heap data.
+    let env = EntryEnv {
+        batch_time: time::GENESIS,
+        canisters_versions_balances_stats: vec![
+            (
+                CanisterId::from_u64(1),
+                0,
+                0_u64.into(),
+                QueryStats::default(),
+            );
+            HEAP_BYTES
+        ],
+    };
+    let env_vec_size = size_of_val(&*env.canisters_versions_balances_stats);
+    let value = EntryValue::new(
+        env,
+        Result::Ok(WasmResult::Reply(vec![42; HEAP_BYTES])),
+        &SystemApiCallCounters::default(),
+    );
+    assert_eq!(size_of_val(&value), 80);
+    assert_eq!(
+        value.memory_bytes(),
+        size_of_val(&value) + env_vec_size + HEAP_BYTES
+    );
+
+    // Value with more heap data.
+    let env = EntryEnv {
+        batch_time: time::GENESIS,
+        canisters_versions_balances_stats: vec![
+            (
+                CanisterId::from_u64(1),
+                0,
+                0_u64.into(),
+                QueryStats::default(),
+            );
+            MORE_HEAP_BYTES
+        ],
+    };
+    let env_vec_size = size_of_val(&*env.canisters_versions_balances_stats);
+    let value = EntryValue::new(
+        env,
+        Result::Ok(WasmResult::Reply(vec![42; MORE_HEAP_BYTES])),
+        &SystemApiCallCounters::default(),
+    );
+    assert_eq!(size_of_val(&value), 80);
+    assert_eq!(
+        value.memory_bytes(),
+        size_of_val(&value) + env_vec_size + MORE_HEAP_BYTES
+    );
```

Comment:
This test is the same as the previous one, just with a different constant: what's the motivation behind repeating it?